### PR TITLE
src/service.c: Use 'input-keyboard' (symbolic) icon as indicator icon on Lomiri.

### PR DESCRIPTION
**Pull request by sunweaver**

### CODE REVIEW

**Review by tari01 — COMMENTED** [review]
If we do this, the indicator will stop indicating the current layout!

**Comment by sunweaver** [comment]
@tari01 That's the intention. At least on Lomiri, the current language icon looks rather clumsy. I will test this patch in MATE tomorrow and I assume that the keyboard icon will be more beautiful there, as well. The current keyboard should be highlighted when opening the indicator's menu.

**Comment by sunweaver** [comment]
Maybe make this configurable through gsettings for MATE et al?

**Comment by sunweaver** [comment]
@tari01 See updated PR branch.

**Review by tari01 — CHANGES_REQUESTED** [review]
GIcon *pIcon = NULL;
    
    if (ayatana_common_utils_is_lomiri())
    {
        m_fnKeyboardGetLayout(self->pPrivate->pKeyboard, -1, &sLanguage, NULL);
        pIcon = g_themed_icon_new_with_default_fallbacks(ICON_DEFAULT);
    }
    else
    {
        gchar *sLanguage;
        m_fnKeyboardGetLayout(self->pPrivate->pKeyboard, -1, &sLanguage, NULL);

        gchar *sIcon = g_strconcat("ayatana-indicator-keyboard-", sLanguage, NULL);
        g_free(sLanguage);

        pIcon = g_themed_icon_new_with_default_fallbacks(sIcon);
        g_free(sIcon);
    }

**Comment by sunweaver** [comment]
@tari01 Your proposed change doesn't seem to work, e.g. gchar *sLanguage is declared in the else clause whereas it gets used in the upper if clause part. Why do we need sLanguage stuff in both if-and-else-clauses?

**Comment by tari01** [comment]
> @tari01 Your proposed change doesn't seem to work, e.g. gchar *sLanguage is declared in the else clause whereas it gets used in the upper if clause part. Why do we need sLanguage stuff in both if-and-else-clauses?

Aaagh, sorry. Copy-paste error. Just drop the `m_fnKeyboardGetLayout` line from upper part.

File: src/service.c [hunk 1]
```diff
@@ -87,14 +89,22 @@ static GVariant* createHeaderState(IndicatorKeyboardService *self)
     g_variant_builder_add(&cBuilder, "{sv}", "title", g_variant_new_string(_("Keyboard")));
     g_variant_builder_add(&cBuilder, "{sv}", "visible", g_variant_new_boolean(TRUE));
 
-    gchar *sLanguage;
-    m_fnKeyboardGetLayout(self->pPrivate->pKeyboard, -1, &sLanguage, NULL);
+    GIcon *pIcon = null;
```
Comment (tari01) [inline src/service.c:92]: @sunweaver 

Final change: `GIcon *pIcon = NULL;`

**Review by tari01 — CHANGES_REQUESTED** [review]

**Review by tari01 — APPROVED** [review]

**Pull request merged**